explore/src/main/scala/explore/targeteditor/AgsOverlay.scala

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.targeteditor
+
+import cats.syntax.all._
+import crystal.react.ReuseView
+import explore.Icons
+import explore.components.ui.ExploreStyles
+import explore.model.formats._
+import explore.model.reusability._
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.feature.ReactFragment
+import japgolly.scalajs.react.vdom.html_<^._
+import lucuma.ags.AgsAnalysis
+import lucuma.ags.AgsGuideQuality
+import lucuma.ags.GuideStarCandidate
+import lucuma.ui.reusability._
+import react.common.ReactFnProps
+import react.common.implicits._
+import react.semanticui.elements.button.Button
+import react.semanticui.sizes._
+
+import scala.math.BigDecimal.RoundingMode
+
+final case class AgsOverlay(
+  selectedGSIndex:     ReuseView[Int],
+  guideStarCandidates: List[(GuideStarCandidate, AgsAnalysis)]
+) extends ReactFnProps[AgsOverlay](AgsOverlay.component)
+
+object AgsOverlay {
+  type Props = AgsOverlay
+
+  // This is used for fov thus needs to be fairly smal
+  implicit val propsReuse: Reusability[Props] = Reusability.derive
```

Comment:
Shall we stick to not using `Reusability` unless necessary?

Reply:
Yes, though is some cases they are needed for the hooks